src/librustc_resolve/error_reporting.rs

```diff
@@ -409,7 +415,35 @@ impl<'a> Resolver<'a> {
             (Def::Union(..), _) |
             (Def::Variant(..), _) |
             (Def::Ctor(_, _, CtorKind::Fictive), _) if ns == ValueNS => {
-                err.span_label(span, format!("did you mean `{} {{ /* fields */ }}`?", path_str));
```

Comment:
Nit: could you move `let (followed_by_brace, closing_brace) = self.followed_by_brace(span);` to here so it's not evaluated unless necessary.